cmd/deploy.go

```diff
@@ -83,27 +90,22 @@ func postDeploymentRequest(l log.Logger, projectName string, namespace string, j
 	runtime := pb.NewRuntimeServiceClient(conn)
 	adapt := v1handler.NewAdapter(pluginRepo, datastoreRepo)
 
-	if len(conf.GetProjectConfig().Global) == 0 {
-		return errors.New("global config should not be empty")
+	projectSpec := &pb.ProjectSpecification{
+		Name:   projectName,
+		Config: conf.GetProject().Config,
+	}
+	namespaceSpec := &pb.NamespaceSpecification{
+		Name:   namespaceName,
+		Config: conf.GetNamespace().Config,
 	}
 
-	// update project config if needed
-	registerResponse, err := runtime.RegisterProject(deployTimeoutCtx, &pb.RegisterProjectRequest{
-		Project: &pb.ProjectSpecification{
-			Name:   projectName,
-			Config: conf.GetProjectConfig().Global,
-		},
-		Namespace: &pb.NamespaceSpecification{
-			Name:   namespace,
-			Config: conf.GetProjectConfig().Local,
-		},
-	})
-	if err != nil {
-		return errors.Wrap(err, "failed to update project configurations")
-	} else if !registerResponse.Success {
-		return fmt.Errorf("failed to update project configurations, %s", registerResponse.Message)
+	if len(projectSpec.Config) > 0 {
+		// update project and namespace config if needed
+		registerProject(l, runtime, deployTimeoutCtx, projectSpec, namespaceSpec)
+	} else {
+		// update namespace config if needed
+		registerProjectNamespace(l, runtime, deployTimeoutCtx, projectSpec.Name, namespaceSpec)
```

Comment:
not merging, keeping exactly as in yaml.

logic should be in the server side.

register project and namespace should be done separately